### PR TITLE
ENH: Add diverging colormap

### PR DESCRIPTION
**Pull request by fepegar**
Slicer lacked a perceptually uniform diverging colormap. According to [matplotlib](https://matplotlib.org/):
> Diverging: change in lightness and possibly saturation of two different colors that meet in the middle at an unsaturated color; should be used when the information being plotted has a critical middle value, such as topography or when the data deviates around zero.

More info about why perceptually uniform colormaps should be used over older colormaps like Rainbow or Jet:
- [matplotlib colormaps](https://matplotlib.org/users/colormaps.html)
- [Diverging colormaps](http://www.kennethmoreland.com/color-maps/)
- [MATLAB's posts about new perceptually uniform colormaps](http://blogs.mathworks.com/steve/2014/10/13/a-new-colormap-for-matlab-part-1-introduction/)

Some perceptually uniform colormaps have already been added to Slicer before, but they were not diverging: #582 

Some examples:
- Correlation between localization of the electrode contacts in a normalized space and some clinical score (left: Rainbow; middle: ColdToHot; right: new diverging BlueRed):
corr
- Visualization of compression / expansion of a deformation field (Jacobian) after a non-linear registration. When using the new diverging colormap, blue means expansion and red means compression (up: Rainbow; middle: ColdToHot; down: new diverging BlueRed):
jacob

When Rainbow or ColdToHot (similar to Jet) are used:
- It's difficult to mentally map a color with a value
- It's not easy to know where values are close to 0
- There is a lot of green areas, because green take a lot of room in the colormap
- Since yellow takes just a bit of room in the colormap, there seems to be a quick jump between green and red and a yellow line between them

Code used to generate the file:
```
import numpy as np
import matplotlib.pyplot as plt
colors = plt.get_cmap('RdBu')(range(255))
colors = colors[::-1] * 255
colors = colors.astype(np.uint8)
link = 'https://matplotlib.org/users/colormaps.html'
outputPath = '/tmp/BlueRed.txt'
opacity = 255
with open(outputPath, 'w') as f:
    f.write('# Inspired form matplotlib colormaps: {}\n'.format(link))
    for i, color in enumerate(colors):
        s = '%d %d %d %d %d 255\n' % (i, i, color[0], color[1], color[2])
        f.write(s)
```

### CODE REVIEW

File: Base/Logic/Resources/ColorFiles/BlueRed.txt [hunk 1]
```diff
@@ -0,0 +1,256 @@
+# Inspired form matplotlib colormaps: https://matplotlib.org/users/colormaps.html
```
Comment (jcfr) [inline Base/Logic/Resources/ColorFiles/BlueRed.txt]: form -> from

Comment (jcfr) [inline Base/Logic/Resources/ColorFiles/BlueRed.txt]: Is there any license associated with the matplot lib colormap ?

Comment (fepegar) [inline Base/Logic/Resources/ColorFiles/BlueRed.txt]: Thanks for reviewing.

For the ones I added in #582 the [licence](https://github.com/BIDS/colormap/blob/master/colormaps.py) is [CC0](https://creativecommons.org/share-your-work/public-domain/cc0), as they say [here](https://bids.github.io/colormap/).

**Comment by jcfr** [comment]
Thanks for the contribution :+1:  I  will fix the typo and integrate.

**Comment by jcfr** [comment]
@pieper @lassoan  I renamed the color file from `RedBlue.txt` to `DivergingBlueRed.txt` (it was conflicting with `vtkMRMLFreeSurferProceduralColorNodeBlueRed`)

If there are no objection, I will integrate later this afternoon.

**Comment by jcfr** [comment]
Integrated in [r26139](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26139)

**Comment by fepegar** [comment]
Thanks! Hope it'll be useful.

**Pull request closed without merging**